File: lista1/zadanie2_java/app/src/main/java/com/example/zadanie2_java/MainActivity.java
package com.example.zadanie2_java;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.appcompat.content.res.AppCompatResources;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Locale;
import java.util.Random;


public class MainActivity extends AppCompatActivity {

    private int playerScore = 0;
    private int computerScore = 0;
    private TextView scoreTextView;
    private ImageView playerChoiceImageView;
    private ImageView computerChoiceImageView;

    enum Choice {
        ROCK,
        PAPER,
        SCISSORS
    }

    Choice getRandomChoice() {
        Random random = new Random();
        int choice = random.nextInt(3);
        if (choice == 0) {
            computerChoiceImageView.setImageDrawable(AppCompatResources.getDrawable(this, R.drawable.rock));
            return Choice.ROCK;
        } else if (choice == 1) {
            computerChoiceImageView.setImageDrawable(AppCompatResources.getDrawable(this, R.drawable.paper));
            return Choice.PAPER;
        } else {
            computerChoiceImageView.setImageDrawable(AppCompatResources.getDrawable(this, R.drawable.scissors));
            return Choice.SCISSORS;
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        scoreTextView = findViewById(R.id.score_text);
        playerChoiceImageView = findViewById(R.id.your_choice);
        computerChoiceImageView = findViewById(R.id.computer_choice);
    }

    public void rockButtonClicked(View view) {
        playerChoiceImageView.setImageDrawable(AppCompatResources.getDrawable(this, R.drawable.rock));

        Choice computerChoice = getRandomChoice();
        if (computerChoice == Choice.ROCK) {
            // draw
            Toast.makeText(this, "Draw!", Toast.LENGTH_SHORT).show();
        } else if (computerChoice == Choice.PAPER) {
            // computer wins
            computerScore++;
            scoreTextView.setText(String.format(Locale.getDefault(), "score: %d-%d", playerScore, computerScore));
            Toast.makeText(this, "Computer wins!", Toast.LENGTH_SHORT).show();
        } else {
            // player wins
            playerScore++;
            scoreTextView.setText(String.format(Locale.getDefault(), "score: %d-%d", playerScore, computerScore));
            Toast.makeText(this, "You win!", Toast.LENGTH_SHORT).show();
        }
    }

    public void paperButtonClicked(View view) {
        playerChoiceImageView.setImageDrawable(AppCompatResources.getDrawable(this, R.drawable.paper));

        Choice computerChoice = getRandomChoice();
        if (computerChoice == Choice.ROCK) {
            // player wins
            playerScore++;
            scoreTextView.setText(String.format(Locale.getDefault(), "score: %d-%d", playerScore, computerScore));
            Toast.makeText(this, "You win!", Toast.LENGTH_SHORT).show();
        } else if (computerChoice == Choice.PAPER) {
            // draw
            Toast.makeText(this, "Draw!", Toast.LENGTH_SHORT).show();
        } else {
            // computer wins
            computerScore++;
            scoreTextView.setText(String.format(Locale.getDefault(), "score: %d-%d", playerScore, computerScore));
            Toast.makeText(this, "Computer wins!", Toast.LENGTH_SHORT).show();
        }
    }

    public void scissorsButtonClicked(View view) {
        playerChoiceImageView.setImageDrawable(AppCompatResources.getDrawable(this, R.drawable.scissors));

        Choice computerChoice = getRandomChoice();
        if (computerChoice == Choice.ROCK) {
            // computer wins
            computerScore++;
            scoreTextView.setText(String.format(Locale.getDefault(), "score: %d-%d", playerScore, computerScore));
            Toast.makeText(this, "Computer wins!", Toast.LENGTH_SHORT).show();
        } else if (computerChoice == Choice.PAPER) {
            // player wins
            playerScore++;
            scoreTextView.setText(String.format(Locale.getDefault(), "score: %d-%d", playerScore, computerScore));
            Toast.makeText(this, "You win!", Toast.LENGTH_SHORT).show();
        } else {
            // draw
            Toast.makeText(this, "Draw!", Toast.LENGTH_SHORT).show();
        }
    }
}

File: lista1/zadanie2_kotlin/app/src/main/java/com/example/zadanie2_kotlin/MainActivity.kt
package com.example.zadanie2_kotlin

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.content.res.AppCompatResources
import kotlin.random.Random


class MainActivity : AppCompatActivity() {
    private var playerScore = 0
    private var computerScore = 0
    private lateinit var scoreTextView: TextView
    private lateinit var playerChoiceImageView: ImageView
    private lateinit var computerChoiceImageView: ImageView

    internal enum class Choice {
        ROCK, PAPER, SCISSORS
    }

    private val randomChoice: Choice
        get() {
            return when (Random.nextInt(3)) {
                0 -> {
                    computerChoiceImageView.setImageDrawable(
                        AppCompatResources.getDrawable(
                            this,
                            R.drawable.rock
                        )
                    )
                    Choice.ROCK
                }
                1 -> {
                    computerChoiceImageView.setImageDrawable(
                        AppCompatResources.getDrawable(
                            this,
                            R.drawable.paper
                        )
                    )
                    Choice.PAPER
                }
                else -> {
                    computerChoiceImageView.setImageDrawable(
                        AppCompatResources.getDrawable(
                            this,
                            R.drawable.scissors
                        )
                    )
                    Choice.SCISSORS
                }
            }
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        scoreTextView = findViewById(R.id.score_text)
        playerChoiceImageView = findViewById(R.id.your_choice)
        computerChoiceImageView = findViewById(R.id.computer_choice)
    }

    fun rockButtonClicked(view: View?) {
        playerChoiceImageView.setImageDrawable(
            AppCompatResources.getDrawable(
                this,
                R.drawable.rock
            )
        )
        when (randomChoice) {
            Choice.ROCK -> {
                // draw
                Toast.makeText(this, "Draw!", Toast.LENGTH_SHORT).show()
            }
            Choice.PAPER -> {
                // computer wins
                computerScore++
                scoreTextView.text =
                    String.format(
                        "score: %d-%d",
                        playerScore,
                        computerScore
                    )

                Toast.makeText(this, "Computer wins!", Toast.LENGTH_SHORT).show()
            }
            else -> {
                // player wins
                playerScore++
                scoreTextView.text =
                    String.format(
                        "score: %d-%d",
                        playerScore,
                        computerScore

                    )
                Toast.makeText(this, "You win!", Toast.LENGTH_SHORT).show()
            }
        }
    }

    fun paperButtonClicked(view: View?) {
        playerChoiceImageView.setImageDrawable(
            AppCompatResources.getDrawable(
                this,
                R.drawable.paper
            )
        )
        when (randomChoice) {
            Choice.ROCK -> {
                // player wins
                playerScore++
                scoreTextView.text =
                    String.format(
                        "score: %d-%d",
                        playerScore,
                        computerScore
                    )
                Toast.makeText(this, "You win!", Toast.LENGTH_SHORT).show()
            }
            Choice.PAPER -> {
                // draw
                Toast.makeText(this, "Draw!", Toast.LENGTH_SHORT).show()
            }
            else -> {
                // computer wins
                computerScore++
                scoreTextView.text =
                    String.format(
                        "score: %d-%d",
                        playerScore,
                        computerScore
                    )

                Toast.makeText(this, "Computer wins!", Toast.LENGTH_SHORT).show()
            }
        }
    }

    fun scissorsButtonClicked(view: View?) {
        playerChoiceImageView.setImageDrawable(
            AppCompatResources.getDrawable(
                this,
                R.drawable.scissors
            )
        )
        when (randomChoice) {
            Choice.ROCK -> {
                // computer wins
                computerScore++
                scoreTextView.text =
                    String.format(
                        "score: %d-%d",
                        playerScore,
                        computerScore
                    )
                Toast.makeText(this, "Computer wins!", Toast.LENGTH_SHORT).show()
            }
            Choice.PAPER -> {
                // player wins
                playerScore++
                scoreTextView.text =
                    String.format(
                        "score: %d-%d",
                        playerScore,
                        computerScore

                    )
                Toast.makeText(this, "You win!", Toast.LENGTH_SHORT).show()
            }
            else -> {
                // draw
                Toast.makeText(this, "Draw!", Toast.LENGTH_SHORT).show()
            }
        }
    }
}